web/tests/narrow_activate.test.js

```diff
@@ -211,4 +212,26 @@ run_test("basics", () => {
     });
 
     assert.equal(narrow_state.narrowed_to_pms(), true);
+
+    message_lists.current.selected_id = () => -1;
+
+    all_messages_data.all_messages_data = {
+        all_messages: () => messages,
+        visibly_empty: () => false,
+        first: () => ({id: 900}),
+        last: () => ({id: 1100}),
+    };
+
+    message_fetch.load_messages_for_narrow = (opts) => {
+        assert.deepEqual(opts, {
+            cont: opts.cont,
+            msg_list: opts.msg_list,
+            anchor: "date",
+            anchor_date: "2023-06-02",
+        });
+
+        opts.cont({anchor: 55}, {anchor: "date"});
+    };
+
+    narrow.activate([{operator: "date", operand: "2023-06-02"}], {trigger: "search"});
```

Comment:
This test case, does cover the new added lines: https://github.com/zulip/zulip/pull/25677/files#diff-6c6eeaa26f7145a3d41a485431ee2442ca53a7aded6a4fdbc265e9575326834dR796-R800. After investigating I found that the test is returning from: https://github.com/zulip/zulip/pull/25677/files#diff-6c6eeaa26f7145a3d41a485431ee2442ca53a7aded6a4fdbc265e9575326834dR786-R789 even though I have set `visibly_empty: () => false` in the mock.